SmartPointers/shared_pointer.hpp

```diff
@@ -1,80 +1,173 @@
-//
-//  shared_pointer.hpp
-//  Pointer
-//
-//  Created by Srushti Desai on 12/27/23.
-//
-
-#ifndef shared_pointer_hpp
```

Comment:
Why did you remove ifdef ? And comments ?

Reply:
Added back

Reply:
It was not added in latest commit

Reply:
fixed